aio/entrypoint.sh

```diff
@@ -57,29 +57,33 @@ function detect_gpu() {
 }
 
 function detect_gpu_size() {
-    if [ "$GPU_ACCELERATION" = true ]; then
-        GPU_SIZE=gpu-8g
-    fi
-
     # Attempting to find GPU memory size for NVIDIA GPUs
-    if echo "$gpu_model" | grep -iq nvidia; then
+    if [ "$GPU_ACCELERATION" = true ] && [ "$GPU_VENDOR" = "nvidia" ]; then
         echo "NVIDIA GPU detected. Attempting to find memory size..."
-        nvidia_sm=($(nvidia-smi --query-gpu=memory.total --format=csv,noheader,nounits))
+        # Using head -n 1 to get the total memory of the 1st NVIDIA GPU detected.
+        # If handling multiple GPUs is required in the future, this is the place to do it
+        nvidia_sm=$(nvidia-smi --query-gpu=memory.total --format=csv,noheader,nounits | head -n 1)
         if [ ! -z "$nvidia_sm" ]; then
-            echo "Total GPU Memory: ${nvidia_sm[0]} MiB"
+            echo "Total GPU Memory: $nvidia_sm MiB"
+            # if bigger than 8GB, use 16GB
+            #if [ "$nvidia_sm" -gt 8192 ]; then
+            #    GPU_SIZE=gpu-16g
+            #else
+            GPU_SIZE=gpu-8g
+            #fi
         else
-            echo "Unable to determine NVIDIA GPU memory size."
+            echo "Unable to determine NVIDIA GPU memory size. Falling back to CPU."
+            GPU_SIZE=cpu
```

Comment:
I'd just go with the minimum gpu setup at this point (we have already detected the gpu at this point)

```suggestion
            GPU_SIZE=gpu-8g
```